module-2/lab-probability-distribution/your_code/main.ipynb

```diff
@@ -12,11 +12,23 @@
   },
```

Comment:
- Compare the distributions above. What do you observe? Explain with the Central Limit Theorem. Has not been completed? What did it happen? 
Take a look on the following: 
Interpretation of the CLT: take a look on these 2 articles (the most clear explanations I've found). You can replicate in Python some of the examples they include.
https://towardsdatascience.com/understanding-the-central-limit-theorem-642473c63ad8.
https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/normal-distributions/central-limit-theorem-definition-examples/
- Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
- Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
- There are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources.
- En estadística no decimos "La hipótesis la rechazamos" or "La hipótesis es aceptada", más bien decimos no podemos rechazar la hipótesis nula cuando corremos un test de contraste de hipótesis y los estadísticos de contraste están dentro del rango que le hemos establecido. Esto lo decimos porque a pesar de un resultado positivo del estadístico de contraste podría no ser normal. Esto lo ves claramente en el primer histograma tamaño de la muestra 5, el p-value te dice que no puedes rechazar la hipótesis nula de normalidad y, sin embargo, el histograma muestra algo que no tiene nada que ver con una normal. Aquí podríamos graficar un gráfico de quantiles y, probablemente, veríamos, al igual que en el histograma, que los datos no vienen de una distribución normal. Lo puedes ver graficando un QQ plot. 
- Regarding this: "What did you notice from the distribution plots? Comment your findings.Muestra una distibución sesgada a la derecha." It's partially true, the main finding is that as larger the sample size the more it seems to be a normal distribution. Take a look on the following and feel free to deepen the concepts.
Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking:
Binomial: X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. We must use a continuity correction.
- Take a look on the plot for poisson, it looks like a normal distribution, don't you think so?